openfecli/commands/gather_abfe.py

```diff
@@ -0,0 +1,301 @@
+import os
+import pathlib
+import sys
+import warnings
+from typing import List, Literal
+
+import click
+import gufe
+import numpy as np
+import pandas as pd
+from openff.units import unit
+
+from openfecli import OFECommandPlugin
+from openfecli.clicktypes import HyphenAwareChoice
+from openfecli.commands.gather import (
+    _collect_result_jsons,
+    format_df_with_precision,
+    load_json,
+    rich_print_to_stdout,
+)
+
+
+def _get_name(result: dict) -> str:
+    """Get the ligand name from a unit's results data.
+
+    Parameters
+    ----------
+    result : dict
+        A results dict.
+
+    Returns
+    -------
+    str
+        Ligand name corresponding to the results.
+    """
+
+    # TODO: is it faster to only load the ligands, or is loading the entire pur cheap?
+    solvent_data = list(result["protocol_result"]["data"]["solvent"].values())[0][0]
+    pur_solvent = gufe.ProtocolUnitResult.from_dict(solvent_data)
+    name = pur_solvent.inputs["alchemical_components"]["stateA"][0].name
```

Comment:
Rather than having to go through the cost of loading the protocol unit result, how about:

```suggestion
    name = solvent_data["inputs"]["alchemical_components"]["stateA"][0]["molprops"]["ofe-name"]
```
In a similar way to how you fetch the name from the ligand atom mapping?